drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -116,217 +121,194 @@ impl Handler {
-                    }
-                }
-
-                MsgType::UpdAbortUpdateReq => {
-                    match self.update.abort_update() {
-                        Ok(()) => {
-                            let rsp = UpdateRspHeader::new(None, None);
-                            hubpack::serialize(tx_payload, &rsp)
-                                .map_or(Err(MsgError::Serialization), |size| {
-                                    Ok((MsgType::UpdAbortUpdateRsp, size))
-                                })
-                        }
-                        Err(_err) => Err(MsgError::Serialization),
-                    }
-                }
-                MsgType::UpdFinishImageUpdateReq => {
-                    match self.update.finish_image_update() {
-                        Ok(()) => {
-                            let rsp = UpdateRspHeader::new(None, None);
-                            hubpack::serialize(tx_payload, &rsp).map_or(
-                                Err(MsgError::Serialization),
-                                |size| {
-                                    Ok((MsgType::UpdFinishImageUpdateRsp, size))
-                                },
-                            )
-                        }
-                        Err(_err) => Err(MsgError::Serialization),
-                    }
-                }
-                MsgType::UpdCurrentVersionReq => {
-                    match self.update.current_version() {
-                        Ok(rsp) => hubpack::serialize(tx_payload, &rsp)
-                            .map_or(Err(MsgError::Serialization), |size| {
-                                Ok((MsgType::UpdCurrentVersionRsp, size))
-                            }),
-                        Err(_err) => Err(MsgError::Serialization),
-                    }
-                }
-                MsgType::SinkReq => {
-                    // The first two bytes of a SinkReq payload are the U16
-                    // mod 2^16 sequence number.
-                    tx_payload[0..2].copy_from_slice(&rx_payload[0..2]);
-                    Ok((MsgType::SinkRsp, 2))
-                }
-                // All of the unexpected messages
-                MsgType::Invalid
-                | MsgType::EchoRsp
-                | MsgType::ErrorRsp
-                | MsgType::SinkRsp
-                | MsgType::SprocketsRsp
-                | MsgType::StatusRsp
-                | MsgType::UpdBlockSizeRsp
-                | MsgType::UpdPrepImageUpdateRsp
-                | MsgType::UpdWriteOneBlockRsp
-                | MsgType::UpdAbortUpdateRsp
-                | MsgType::UpdFinishImageUpdateRsp
-                | MsgType::UpdCurrentVersionRsp
-                | MsgType::Unknown => {
-                    status.rx_invalid = status.rx_invalid.wrapping_add(1);
-                    Err(MsgError::BadMessageType)
-                }
-            }
-        };
         // The above cases either enqueued a message and returned size
```

Comment:
This comment is out of date now, I think. Now it's referring to what `self.run()` returns?

Reply:
Comment removed in [f86812d](https://github.com/oxidecomputer/hubris/pull/965/commits/f86812d56e41435275dab8b4cf32982f88d4c35f)